src/ArcherContainer.js

```diff
@@ -298,36 +403,24 @@ export class ArcherContainer extends React.Component<Props, State> {
    * a different color or size
    * */
   _generateAllArrowMarkers = (): React$Element<'marker'>[] => {
-    return this._getSourceToTargets().map(
-      ({ source, target, label, style }: SourceToTargetType) => {
-        const strokeColor = (style && style.strokeColor) || this.props.strokeColor;
-
-        // Actual arrowLength value might be 0, which can't work with a simple 'actualValue || defaultValue'
-        let arrowLength = this.props.arrowLength;
-        if (style && (style.arrowLength || style.arrowLength === 0)) {
-          arrowLength = style.arrowLength;
-        }
-
-        const arrowThickness = (style && style.arrowThickness) || this.props.arrowThickness;
-
-        const arrowPath = `M0,0 L0,${arrowThickness} L${arrowLength},${arrowThickness / 2} z`;
```

Comment:
Rather than add all the logic for `circle` in this function, I've moved all of this up into `_buildShape`. This would also make adding more shapes in the future easier & more organized.